source/agora/node/FullNode.d

```diff
@@ -107,8 +107,14 @@ public class FullNode : API
     /// Task manager
     protected ITaskManager taskman;
 
-    /// Timer this node has started
-    protected ITimer[] timers;
+    protected enum FullNodeTimers
+    {
+        Discovery,
+        BlockCatchup,
+        ClockTick
+    }
+    /// Timers this node has started
+    protected ITimer[int] timers;
```

Comment:
Few things:
1) I'm not sure an AA is better. AA provide amortized O(1) access, but `timers` are only written to during initialization, so nothing can beat arrays which offer guaranteed O(1) access. The only thing which would make it better is a static array because it'd avoid bounds check.
2) Trailing comma :)

Reply:
Also I would just name them `TimerIdx` or something. They are already namespaced by being in the `FullNode` class.